pkg/apis/hive/v1/validating-webhooks/syncset_validating_admission_hook.go

```diff
@@ -247,6 +256,14 @@ func validatePatches(patches []hivev1.SyncObjectPatch, fldPath *field.Path) fiel
 	return allErrs
 }
 
+func validateResourceApplyMode(resourceApplyMode hivev1.SyncSetResourceApplyMode, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if resourceApplyMode != "" && !validResourceApplyModes[string(resourceApplyMode)] {
```

Comment:
If the keys of `resourceApplyMode` are `SyncSetResourceApplyMode` objects, then you do not need the cast to `string`.